x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -358,31 +369,56 @@ export const entryHasListType = (
  * Returns the value for `file.Ext.code_signature` which
  * can be an object or array of objects
  */
-export const getCodeSignatureValue = (
-  alertData: Ecs
+export const getFileCodeSignature = (
+  alertData: FlattenType<Ecs>
 ): Array<{ subjectName: string; trusted: string }> => {
   const { file } = alertData;
   const codeSignature = file && file.Ext && file.Ext.code_signature;
 
-  // Pre 7.10 file.Ext.code_signature was mistakenly populated as
-  // a single object with subject_name and trusted.
+  return getCodeSignatureValue(codeSignature);
+};
+
+/**
+ * Returns the value for `process.Ext.code_signature` which
+ * can be an object or array of objects
+ */
+export const getProcessCodeSignature = (
+  alertData: FlattenType<Ecs>
+): Array<{ subjectName: string; trusted: string }> => {
+  const { process } = alertData;
+  const codeSignature = process && process.Ext && process.Ext.code_signature;
+
+  return getCodeSignatureValue(codeSignature);
+};
+
+/**
+ * Pre 7.10 `Ext.code_signature` fields were mistakenly populated as
+ * a single object with subject_name and trusted.
+ */
+export const getCodeSignatureValue = (
+  codeSignature: FlattenType<CodeSignature> | FlattenType<CodeSignature[]> | undefined
+): Array<{ subjectName: string; trusted: string }> => {
   if (Array.isArray(codeSignature) && codeSignature.length > 0) {
-    return codeSignature.map((signature) => ({
-      subjectName: (signature.subject_name && signature.subject_name[0]) ?? '',
-      trusted: (signature.trusted && signature.trusted[0]) ?? '',
-    }));
+    return codeSignature.map((signature) => {
+      const subjectName = signature.subject_name ?? '';
+      const trusted = signature.trusted ?? '';
+
+      return {
+        subjectName: Array.isArray(subjectName) ? subjectName[0] : subjectName,
```

Comment:
Is there a way to combine this logic with the logic in line [403](https://github.com/elastic/kibana/pull/89974/files#diff-6808c52ab81432d126e55b4a4d4ff277c9fb83276e3f9170ca8f0ae1ce922efcR403)? Right now the definition of `subjectName` and the definition of `trusted` is split between two lines.